### PR TITLE
NBS-4814: [Disk Manager] intoduce vhd image_map_reader; implement ReadHeader and ReadFooter 

### PR DESCRIPTION
**Pull request by BarkovBG**

### CODE REVIEW

**Comment by BarkovBG** [comment]
https://github.com/ydb-platform/nbs/issues/498

File: cloud/disk_manager/internal/pkg/dataplane/url/vhd/image_map_reader.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	if string(r.footer.Cookie[:]) != footerCookie {
+		return common.NewSourceInvalidError(
+			"Failed to check vhd footer cookie: expected - %s, actual - %s",
+			footerCookie,
+			r.footer.Cookie,
+		)
+	}
+
+	if !r.footer.validate() {
+		return common.NewSourceInvalidError(
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/dataplane/url/vhd/image_map_reader.go:80]: Lets move this to footer.validate() func, making it return an error

File: cloud/disk_manager/internal/pkg/dataplane/url/vhd/image_map_reader.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+		offset,
+		headerSize,
+		binary.LittleEndian,
+		&r.header,
+	)
+	if err != nil {
+		return err
+	}
+
+	if string(r.header.Cookie[:]) != headerCookie {
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/dataplane/url/vhd/image_map_reader.go:109]: Lets move it into validate func

File: cloud/disk_manager/internal/pkg/dataplane/url/vhd/image_map_reader.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	return r.readHeader(ctx, 512)
+}
+
+func (r *ImageMapReader) Read(context.Context) ([]common.ImageMapItem, error) {
+	return []common.ImageMapItem{}, nil // TODO: Implement.
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+func (r *ImageMapReader) readFooter(ctx context.Context) error {
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/dataplane/url/vhd/image_map_reader.go:50]: This should be called from Read function

File: cloud/disk_manager/internal/pkg/dataplane/url/vhd/image_map_reader.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+	if !r.footer.validate() {
+		return common.NewSourceInvalidError(
+			"Failed to check vhd footer validity: footer is corrupted",
+		)
+	}
+
+	return nil
+}
+
+func (r *ImageMapReader) readHeader(ctx context.Context, offset uint64) error {
```
Comment (SvartMetal) [inline cloud/disk_manager/internal/pkg/dataplane/url/vhd/image_map_reader.go:88]: Hardcode offset here

**Review by leftmain — APPROVED** [review]

**Pull request merged**